src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs

```diff
@@ -152,8 +153,12 @@ public GetValueOrDefault(MethodInfo mi)
                 _defaultValueType = mi.ReturnType;
             }
 
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2077:UnrecognizedReflectionPattern",
+                Justification = "_defaultValueType is a ValueType, so it will always have a default constructor.")]
             public override int Run(InterpretedFrame frame)
             {
+                Debug.Assert(_defaultValueType.IsValueType, "Nullable only allows on value types.");
+
```

Comment:
Nit: It would be cleaner to put the suppression and assert onto the .ctor of this class - since that's the source of the "knowledge". And then just annotate the field with the right DAM.